src/issue.rs

```diff
@@ -31,82 +30,525 @@ fn issue_actor_and_project(
+				})
+			}) {
+			db_entry.issue_project_state =
+				Some(ProjectState::Confirmed(backlog_column.id));
+			github_bot.create_project_card(
+				backlog_column.id,
+				issue.id.context(error::MissingData)?,
+				github::ProjectCardContentType::Issue,
+			)?;
+			DbEntryState::Update
+		} else {
+			// TODO warn that the project lacks a backlog column
+			DbEntryState::DoNothing
+		},
+	)
+}
+
+fn author_core_no_project(
+	db_entry: &mut DbEntry,
 	github_bot: &GithubBot,
 	matrix_bot: &MatrixBot,
-	core_devs: &[github::User],
-	github_to_matrix: &HashMap<String, String>,
-	projects: Option<&project::Projects>,
 	issue: &github::Issue,
-) -> Result<()> {
-	// TODO: handle multiple projcets in a single repo
-	let project_info =
-		projects.and_then(|p| p.0.iter().last().map(|p| p.1.clone()));
+	default_channel_id: &str,
+	since: Option<Duration>,
+) -> Result<DbEntryState> {
+	let issue_id = issue.id.context(error::MissingData)?;
+	let issue_html_url = issue.html_url.as_ref().context(error::MissingData)?;
 
-	let author = &issue.user;
-	let repo = &issue.repository;
-	let author_info = project_info
-		.map_or_else(project::AuthorInfo::default, |p| {
-			p.author_info(&author.login)
-		});
-	let author_is_core = core_devs.iter().any(|u| u.id == author.id);
+	let ticks = since
+		.map(|elapsed| elapsed.as_secs() / ISSUE_NO_PROJECT_CORE_PING_PERIOD);
 
+	Ok(match ticks {
+		None => {
+			db_entry.issue_no_project_ping = Some(SystemTime::now());
+			matrix_bot.send_public_message(
+				default_channel_id,
+				&ISSUE_NO_PROJECT_MESSAGE.replace("{1}", issue_html_url),
+			)?;
+			DbEntryState::Update
+		}
+		Some(0) => DbEntryState::DoNothing,
+		Some(i) => {
+			if i == ISSUE_NO_PROJECT_ACTION_AFTER_NPINGS {
+				// If after 3 days there is still no project
+				// attached, move the issue to Core Sorting
+				// repository
+				github_bot.close_issue(&issue.repository.name, issue_id)?;
+				let params = serde_json::json!({
+						"title": issue.title,
+						"body": issue.body.as_ref().unwrap_or(&"".to_owned())
+				});
+				github_bot.create_issue(CORE_SORTING_REPO, params)?;
+				DbEntryState::Delete
+			} else if (db_entry.issue_no_project_npings) < i {
+				db_entry.issue_no_project_npings = i;
+				matrix_bot.send_public_message(
+					default_channel_id,
+					&ISSUE_NO_PROJECT_MESSAGE.replace("{1}", issue_html_url),
+				)?;
+				DbEntryState::Update
+			} else {
+				DbEntryState::DoNothing
+			}
+		}
+	})
+}
+
+fn author_unknown_no_project(
+	db_entry: &mut DbEntry,
+	github_bot: &GithubBot,
+	matrix_bot: &MatrixBot,
+	issue: &github::Issue,
+	default_channel_id: &str,
+	since: Option<Duration>,
+) -> Result<DbEntryState> {
 	let issue_id = issue.id.context(error::MissingData)?;
 	let issue_html_url = issue.html_url.as_ref().context(error::MissingData)?;
 
-	match issue_actor_and_project(issue, github_bot)? {
+	let ticks = since.map(|elapsed| {
+		elapsed.as_secs() / ISSUE_NO_PROJECT_NON_CORE_PING_PERIOD
+	});
+
+	Ok(match ticks {
```

Comment:
Like the above function I think whether `ticks` is `None` or not should be handled separately. It also seems that what you're doing in the `None` branch is same across a few of the functions. Maybe it could be extracted out to a function?